File: koopey.web/src/old/models/bitcoin.ts
export class Bitcoin {
    public account: string = "";
    public address: string = "";
    public amount: number = 0;
    public asm: string = "";
    public balance: number;
    public bestblockhash: string = "";
    public bip9_softforks: Array<any>;
    public blocks: number = 0;
    public chain: string = "";
    public chainwork: string; 
    public complete: Boolean = false;
    public connections: number
    public confirmations: number = 0;  
    public difficulty: number = 0;
    public fromaddress: string = "";
    public fromaccount: string = "";
    public hash: string = "";
    public headers: number = 0;
    public hex: string = "";
    public iscompressed: Boolean = false;
    public isvalid: Boolean = false;
    public ismine: Boolean = false;
    public iswatchonly: Boolean = false;
    public isscript: Boolean = false;
    public keypoololdest: number;
    public keypoolsize: number;
    public localaddresses: Array<any>
    public localrelay: boolean
    public localservices: string   
    public mediantime: number = 0;
    public networks: Array<any        >
    public type: string = "";
    public txid: string = "";
    public protocolversion : number
    public pruned: boolean = false;
    public pruneheight: number = 0; 
    public pubkey: string = "";
    public relayfee: number
    public scriptPubKey: string = "";
    public softforks: Array<any>;   
    public timeoffset: number
    public timestamp: number = 0;  
    public toaddress: string = "";
    public txcount: number;
    public spendable: Boolean = false;
    public solvable: Boolean = false;
    public subversion: string;  
    public unlocked_until: number;
    public verificationprogress: number = 0;
    public version: number = 0;
    public vout: number = 0;
    public warnings: string
    public walletversion: number   
}

File: koopey.android/app/src/main/java/com/koopey/model/Bitcoin.java
package com.koopey.model;

import com.koopey.model.base.Base;

import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.SuperBuilder;

@Data
@EqualsAndHashCode(callSuper = true)
@SuperBuilder
public class Bitcoin extends Base {

    public static final String BITCOIN_FILE_NAME = "bitcoin.dat";
    private String account;
    private String address;
    private String asm;
    private String fromaccount;
    private String fromaddress;
    private String hash;
    private String hex;
    private String pubkey;
    private String scriptPubKey;
    private String toaddress;
    private String txid;

    @Builder.Default
    private Double amount = 0d;
    @Builder.Default
    private int confirmations = 0;
    @Builder.Default
    private int timestamp = 0;
    @Builder.Default
    private int vout = 0;
    @Builder.Default
    private int version = 0;
    @Builder.Default
    private boolean complete = true;
    @Builder.Default
    private boolean iscompressed = false;
    @Builder.Default
    private boolean isvalid = false;
    @Builder.Default
    private boolean ismine = false;
    @Builder.Default
    private boolean iswatchonly = false;
    @Builder.Default
    private boolean isscript = false;
    @Builder.Default
    private boolean spendable = true;
    @Builder.Default
    private boolean solvable = true;

    public Bitcoin() {
    }

    public boolean isBalance() {
        return this.getType().equals("balance");
    }

    public boolean isTransaction() {
        return this.getType().equals("transaction");
    }


    public boolean isEmpty() {
        return (this.account.isEmpty() && this.address.isEmpty()) ? true : false;
    }

    public void parseWallet(Wallet wallet) {
        this.address = wallet.getName();
    }

}
